src/qt/transactionrecord.cpp

```diff
@@ -436,36 +436,38 @@ bool IsZPIVType(TransactionRecord::Type type)
 void TransactionRecord::updateStatus(const CWalletTx& wtx)
 {
     AssertLockHeld(cs_main);
-    // Determine transaction status
+    int chainHeight = chainActive.Height();
 
+    CBlockIndex *pindex = nullptr;
     // Find the block the tx is in
-    CBlockIndex* pindex = NULL;
     BlockMap::iterator mi = mapBlockIndex.find(wtx.hashBlock);
     if (mi != mapBlockIndex.end())
         pindex = (*mi).second;
 
+    // Determine transaction status
+
     // Sort order, unrecorded transactions sort to the top
     status.sortKey = strprintf("%010d-%01d-%010u-%03d",
         (pindex ? pindex->nHeight : std::numeric_limits<int>::max()),
         (wtx.IsCoinBase() ? 1 : 0),
         wtx.nTimeReceived,
         idx);
-    //status.countsForBalance = wtx.IsTrusted() && !(wtx.GetBlocksToMaturity() > 0);
-    bool fConflicted;
-    status.depth = wtx.GetDepthAndMempool(fConflicted);
-    const bool isOffline = (GetAdjustedTime() - wtx.nTimeReceived > 2 * 60 && wtx.GetRequestCount() == 0);
 
-    //Determine the depth of the block
-    int nBlocksToMaturity = wtx.GetBlocksToMaturity();
+    bool fConflicted = false;
+    int depth = 0;
+    bool isTrusted = wtx.IsTrusted(depth, fConflicted);
+    const bool isOffline = (GetAdjustedTime() - wtx.nTimeReceived > 2 * 60 && wtx.GetRequestCount() == 0);
+    int nBlocksToMaturity = (wtx.IsCoinBase() || wtx.IsCoinStake()) ? std::max(0, (Params().COINBASE_MATURITY() + 1) - depth) : 0;
```

Comment:
this is exaclty the same as `wtx.GetBlocksToMaturity()`, why replace it?

Reply:
`GetBlocksToMaturity` calls `GetDepthInMainChain` which is what `IsTrusted` is calling inside the `GetDepthAndMempool` method call.

It's in the PR description, second point. The same `GetDepthInMainChain` method was called in three different ways with, ofc, the same result. But more importantly,  requires `cs_main` lock which should occur as less as possible in any case.
 Aiming to, in the future, keep improving the `updateStatus` method to not depend on the lock.

Reply:
Ok